cmd/kube-proxy/app/server.go

```diff
@@ -709,6 +710,13 @@ func (s *ProxyServer) Run() error {
 				return err
 			}
 		}
+
+		// check if the conntrack user-space tool is installed, it's needed by the kube-proxy mainly for clearing
+		// and flushing conntrack entries that remain active after a service is deleted and may blackhole traffic
+		if !conntrack.Exists(s.execer) {
+			const message = "The binary conntrack is not installed, this can cause failures in network connection cleanup."
```

Comment:
"The conntrack binary" sounds better than "The binary conntrack" to me.

I'm not sure if an Event is enough here... saying "it causes failures in cleanup" makes it sound like the problem is just that stale state may be left around, but it's not that; connections might be mis-routed after a service change, causing connections to fail. And unless you know a lot about kube-proxy, it's not obvious that the reason they're failing is because a binary is missing.

Hm... the netlink package we already depend on has [code to send conntrack-related messages](https://github.com/vishvananda/netlink/blob/master/conntrack_linux.go#L89). Maybe the right longer-term fix is to just port `pkg/util/conntrack` to not need the binary.

Reply:
> the netlink package we already depend on has code to send conntrack-related messages. > Maybe the right longer-term fix is to just port pkg/util/conntrack to not need the binary.

that's, definitely, the right fix ... will check how difficult it is